### ISSUE TITLE
Project Tree View Drag & Drop Synchronization (2.6b1)

### ISSUE THREAD

**Issue by awqk** [vkbo/novelWriter#2137]
It's possible to move a tree view item (We Found John!) to the root of the tree view, see screenshot.
Any attempt to put it back fails (see screenshot).
Close and Reopen seems to fix things.

I also managed this for once for a subtree, but failed to reproduce this.

![treeviewbug](https://github.com/user-attachments/assets/965322d6-cd9f-4a9f-ba0a-8e38d38eaf7c)

**Comment by vkbo** [comment]
Yes, this is a bug that was present in 2.5.2 and before as well. It is a duplicate of ticket #2108. It was fixed in 2.5.3. It can be triggered by dragging an item onto the indented space on the left of another item, which is interpreted as a viewport drop. It was fixed in #2109. It only affects Linux .deb installations on platforms running on older Qt versions.

It is also present in 2.6b1 as you've noticed, but the entire code behind this has been removed afterwards, so the bug is no longer relevant. The entire project tree has been re-implemented using a custom data model that blocks these kind of moves on the data level instead of relying on intercepting the drag and drop action to cancel it.

I plan to make a 2.6b2 release before the holidays.

TLDR: Closing this as it's already fixed in 2.5.x, and irrelevant for 2.6.x.
